src/components/app/routes/data/services.js

```diff
@@ -0,0 +1,266 @@
+ * queries
+ * .enterprise
+ * .enterpriseCustomer(enterpriseUuid)
+ * ._ctx.contentHighlights
+ * ._ctx.configuration
+ * @returns
+ */
+export function queryContentHighlightsConfiguration(enterpriseUuid) {
+  return queries
+    .enterprise
+    .enterpriseCustomer(enterpriseUuid)
+    ._ctx.contentHighlights
+    ._ctx.configuration;
+}
+// 'contentHighlights' contextQueries END
+
+// 'course' context layer START
+// 'course' contextQueries START
+/**
+ * Helper function to assist querying with useQuery package
+ * queries
+ * .enterprise
+ * .enterpriseCustomer(enterpriseUuid)
+ * ._ctx.course
+ * ._ctx.contentMetadata(courseKey)
+ * @returns
+ */
+export function queryCourseMetadata(enterpriseUuid, courseKey) {
+  return queries
+    .enterprise
+    .enterpriseCustomer(enterpriseUuid)
+    ._ctx.course
+    ._ctx.contentMetadata(courseKey);
+}
+
+/**
+ * Helper function to assist querying with useQuery package
+ * queries
+ * .enterprise
+ * .enterpriseCustomer(enterpriseUuid)
+ * ._ctx.course
+ * ._ctx.canRedeem(availableCourseRunKeys)
+ * @returns
+ */
+export function queryCanRedeem(enterpriseUuid, courseMetadata) {
+  const availableCourseRunKeys = getAvailableCourseRuns(courseMetadata).map(courseRun => courseRun.key);
+  return queries
+    .enterprise
+    .enterpriseCustomer(enterpriseUuid)
+    ._ctx.course
+    ._ctx.canRedeem(availableCourseRunKeys);
+}
+// 'course' contextQueries END
+// 'course' context layer START
+
+// 'enrollments' context layer START
+/**
+ * Helper function to assist querying with useQuery package
+ * queries
+ * .enterprise
+ * .enterpriseCustomer(enterpriseUuid)
+ * ._ctx.enrollments
+ * @returns
+ */
+export function queryEnterpriseCourseEnrollments(enterpriseUuid) {
+  return queries
+    .enterprise
+    .enterpriseCustomer(enterpriseUuid)
+    ._ctx.enrollments;
+}
+// 'enrollments' context layer END
+
+// 'subsidies' context layer START
+// 'subsidies' contextQueries START
+
+// 'browseAndRequest' context layer START
+// 'browseAndRequest' contextQueries START
+/**
+ * Helper function to assist querying with useQuery package
+ * queries
+ * .enterprise
+ * .enterpriseCustomer(enterpriseUuid)
+ * ._ctx.subsidies
+ * ._ctx.browseAndRequest(userEmail)
+ * ._ctx.configuration
+ * @returns
+ */
+export function queryBrowseAndRequestConfiguration(enterpriseUuid, userEmail) {
+  return queries
+    .enterprise
+    .enterpriseCustomer(enterpriseUuid)
+    ._ctx.subsidies
+    ._ctx.browseAndRequest(userEmail)
+    ._ctx.configuration;
+}
+
+// 'endpoints' context layer START
+// 'endpoint contextQueries START
+/**
+ * Helper function to assist querying with useQuery package
+ * queries
+ * .enterprise
+ * .enterpriseCustomer(enterpriseUuid)
+ * ._ctx.subsidies
+ * ._ctx.browseAndRequest(userEmail)
+ * ._ctx.endpoints(state)
+ * ._ctx.licenseRequests
+ * @returns
+ */
+export function queryLicenseRequests(enterpriseUuid, userEmail, state = SUBSIDY_REQUEST_STATE.REQUESTED) {
```

Comment:
Is anything using these new query definitions for `queryLicenseRequests` and `queryCouponCodeRequests`? It looks like the API calls to fetch license and coupon code requests still occurs within the same query as the B&R configuration.

I believe you'd want to remove those calls to fetch requests from the configuration query fn (i.e., `fetchBrowseAndRequestConfiguration`) and instead include an `ensureQueryData` for each of these new requests-related queries. It might also require an update to the `useEnterpriseCustomerUserSubsidies` hook to also include these specific `queryLicenseRequests` and `queryCouponCodeRequests` as well.